hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestMasterRegistry.java

```diff
@@ -103,7 +104,17 @@ private static String generateDummyMastersList(int size) {
   @Test public void testRegistryRPCs() throws Exception {
     Configuration conf = new Configuration(TEST_UTIL.getConfiguration());
     HMaster activeMaster = TEST_UTIL.getHBaseCluster().getMaster();
-    for (int numHedgedReqs = 1; numHedgedReqs <=3; numHedgedReqs++) {
+    final int size = activeMaster.getMetaRegionLocationCache().
+      getMetaRegionLocations().get().size();
+    // Add wait on all replicas being assigned before proceeding w/ test. Failed on occasion
+    // because not all replicas had made it up before test started.
+    TEST_UTIL.waitFor(10000,
+      () -> {
+        try (MasterRegistry registry = new MasterRegistry(conf)) {
+          return registry.getMetaRegionLocations().get().size() >= size;
```

Comment:
this looks always true since the registry actually gets the count from activeMaster.getMetaRegionLocationCache()?

I think the check should look something like activeMaster.getLocations().size() == NUM_REPLICAS? Actually we have a  utility for this somewhere..I'm not able to find it now..:(

Reply:
You'd know better. There is no getLocations. What you think you were referring to?

Reply:
Found it ..RegionReplicaTestHelper#waitUntilAllMetaReplicasAreReady().. sorry for the typo..the condition should be something like the above method..(I also fixed an edge case there to make the check tighter)..